source/Coop.Core/Client/States/ValidateModuleState.cs

```diff
@@ -64,6 +67,19 @@ internal void Handle_NetworkClientValidated(MessagePayload<NetworkClientValidate
         }
     }
 
+    internal void Handle_NetworkModuleVersionsValidated(MessagePayload<NetworkModuleVersionsValidated> obj)
+    {
+        if (obj.What.Matches)
+        {
+            network.SendAll(new NetworkClientValidate(controllerIdProvider.ControllerId));
+        }
+        else
+        {
+            messageBroker.Publish(this, new SendInformationMessage(obj.What.Reason));
+            Disconnect();
```

Comment:
Does disconnect properly change the state back to MainMenuState?

Reply:
The status is set directly now